src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1111,26 +1111,25 @@ private ImmutableArray<BoundNode> ConvertCollectionExpressionElements(
                 int conversionIndex = 0;
                 foreach (var element in elements)
                 {
+                    if (element is BoundCollectionExpressionWithElement)
+                    {
+                        continue;
+                    }
+                    var elementConversion = elementConversions[conversionIndex++];
```

Comment:
So the `elementConversions` is possibly not the same length as `elements`? Should we assert on the relationship between the lengths?

Reply:
We have an assert after the loop that we've used all conversions.